iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/KubernetesHighlightingVisitor.java

```diff
@@ -0,0 +1,165 @@
+
+  private static final Logger LOG = LoggerFactory.getLogger(KubernetesHighlightingVisitor.class);
+  private static final Pattern LINE_SEPARATOR = Pattern.compile("\\r\\n|[\\n\\r\\u2028\\u2029]");
+
+  private static final String DOUBLE_QUOTED_VALUE_GROUP_NAME = "doubleQuotedValue";
+  private static final String SINGLE_QUOTED_VALUE_GROUP_NAME = "singleQuotedValue";
+  private static final String QUOTELESS_VALUE_GROUP_NAME = "quotelessValue";
+
+  private static final String DOUBLE_QUOTED_KEY = quoted("doubleQuotedKey", "\"");
+  private static final String DOUBLE_QUOTED_VALUE = quoted(DOUBLE_QUOTED_VALUE_GROUP_NAME, "\"");
+  private static final String SINGLE_QUOTED_KEY = quoted("singleQuotedKey", "'");
+  private static final String SINGLE_QUOTED_VALUE = quoted(SINGLE_QUOTED_VALUE_GROUP_NAME, "'");
+  private static final String QUOTELESS_KEY = quoteless("quotelessKey");
+  private static final String QUOTELESS_VALUE = quoteless(QUOTELESS_VALUE_GROUP_NAME);
+  private static final String MULTI_LINE_OPERATORS = "(?<multilineOperator>[|>])";
+  private static final String KEY = DOUBLE_QUOTED_KEY + "|" + SINGLE_QUOTED_KEY + "|" + QUOTELESS_KEY;
+  private static final String VALUE = DOUBLE_QUOTED_VALUE + "|" + SINGLE_QUOTED_VALUE + "|" + QUOTELESS_VALUE;
+
+  // still needs to be tested
+  private static final Pattern YAML_DIRECTIVES = Pattern.compile("\\s*(?:---|%).*");
+
+  private static final String COMMENT_S = "(?:\\h(?<comment>#.*))?";
+  private static final String TAG = "(?<tag>!\\H+\\h?)?";
+  private static final String COMBINED = "\\h*-?\\h*(?:" + KEY + "):(?:\\h+" + TAG + "(?:" +
+    MULTI_LINE_OPERATORS + "|" + VALUE + ")?)?\\h*" + COMMENT_S;
+
+  private static final Pattern COMBINED_PATTERN = Pattern.compile(COMBINED);
+
+  private static final String SCALAR_VALUE = "\\h*-?\\h*" + TAG + "(?:" + MULTI_LINE_OPERATORS + "|" + VALUE + ")?\\h*" + COMMENT_S;
+  private static final Pattern SCALAR_VALUE_PATTERN = Pattern.compile(SCALAR_VALUE);
+
+  private static final Set<String> GROUPNAMES_IN_SCALAR_VALUES = Set.of("tag",
+    "multilineOperator",
+    DOUBLE_QUOTED_VALUE_GROUP_NAME,
+    SINGLE_QUOTED_VALUE_GROUP_NAME,
+    QUOTELESS_VALUE_GROUP_NAME,
+    "comment");
+
+  private static final Map<String, TypeOfText> GROUPNAME_TO_TYPE = Map.of(
+    "doubleQuotedKey", KEYWORD,
+    "singleQuotedKey", KEYWORD,
+    "quotelessKey", KEYWORD,
+    "multilineOperator", KEYWORD_LIGHT,
+    "tag", KEYWORD_LIGHT,
+    DOUBLE_QUOTED_VALUE_GROUP_NAME, STRING,
+    SINGLE_QUOTED_VALUE_GROUP_NAME, STRING,
+    QUOTELESS_VALUE_GROUP_NAME, STRING,
+    "comment", COMMENT);
+
+  private static String quoted(String groupName, String quotes) {
+    return "(?<" + groupName + ">" + quotes + "[^" + quotes + "]*" + quotes + ")";
+  }
+
+  // Current limitation: We don't allow '#' in quoteless variable names
+  private static String quoteless(String groupName) {
+    return "(?<" + groupName + ">[^'\"%#]*)";
+  }
```

Comment:
From my test with online Yaml parser, it seems allowed to have a `'` or `"` in the middle/end of a key/value.
Example:
```
test:
  k'ey1: va'lue
  ke"y2: value"
```
You just cannot have them as the first character (it would be considered as a quoted key/value).
```suggestion
  private static String quoteless(String groupName) {
    return "(?<" + groupName + ">(?:[^'\"%#][^%#]*)?)";
  }
```